runtimes/web/src/index.js

```diff
@@ -245,6 +245,62 @@ async function loadCartWasm () {
     window.addEventListener("keydown", onKeyboardEvent);
     window.addEventListener("keyup", onKeyboardEvent);
 
+    let gamepads = 0;
+    function processGamepad() {
+        if (gamepads === 0) {
+            return;
+        }
+
+        const gamepad = navigator.getGamepads()[0];
+        if (!gamepad) {
```

Comment:
I think there's a bug here where when the gamepad is disconnected and reconnected, it'll stop working because gamepad index 0 will still be null.

Could we store the last gamepad index connected in the `gamepadconnected` handler, and then use it in `processGamepad`?

Reply:
This is problematic case. I have a 2 gamepads and, 1 is real, 2 is virtual.
And a lot of gamepad mapping programms replace a 1 gamepad.
If we will use a latest connected gamepad - it will not working.

Reply:
And looks like that when gamepad is disconnected it changings a gamepad list and it never can be a gaps.

Reply:
We could make it the oldest still connected gamepad index, instead of the most recently connected. It should just be a valid index.

Reply:
Yep, i do this.
I checked a bechaviour, i have a 2 gamepads (BL and usb) and there are a difference.
XBox gamepad is always is 0, but USB is not change a index when XBox is unplugging.

Reply:
Note.
maskGamepad working some strange. Because it use a ` buttons &= ~maks` for rest need use a `maskGamepad(0, ~1, false)`  instead of `maskGamepad(0,0)`